### PR TITLE
add detekt for lint check and add detekt in fastlane check and add gi…

### PR DESCRIPTION
**Pull request by Tony---Zhang**
## Overview
- add `detekt` for lint check
- add `detekt` in fastlane check lane
- add git hooks for check `detekt`
- fix detekt issues

## Type of change
- [x] New feature (non-breaking change which adds functionality)

## Check list
1. [x] Does your submission pass tests?
2. [x] Have you lint your code locally prior to submission?
3. [x] Have you update the wiki of your change?

### CODE REVIEW

File: scripts/githooks/hooks/pre-push [hunk 1]
```diff
@@ -2,5 +2,6 @@
 set -euo pipefail
 
 echo "=====unit test start====="
+bundle exec fastlane check
```
Comment (yunlongwen) [inline scripts/githooks/hooks/pre-push:5]: `lint check` should not be placed in the echo of unit_test

**Comment by ssseasonnn** [comment]
good job

File: app/src/main/java/com/thoughtworks/android/ark/ui/dashboard/DashboardFragment.kt [hunk 2]
```diff
@@ -40,4 +40,4 @@ class DashboardFragment : Fragment() {
         super.onDestroyView()
         _binding = null
     }
-}
\ No newline at end of file
+}
```
Comment (yunlongwen) [inline app/src/main/java/com/thoughtworks/android/ark/ui/dashboard/DashboardFragment.kt:43]: A new line appeared?
`NewLineAtEndOfFile: active: false`

File: scripts/githooks/hooks/pre-push (hunk 1 above)
Comment (Tony---Zhang) [reply scripts/githooks/hooks/pre-push:5]: For `lint check` we will integration it into `fastlane check` lane, for git hook implement, it is OK for now

**Pull request merged**